app/controllers/tags_controller.rb

```diff
@@ -96,6 +96,19 @@ def publish
     end
   end
 
+  def destroy
+    respond_to do |format|
+      format.json do
+        if Artefact.with_tags([@tag.tag_id]).any?
+          render json: { error: 'Tag has documents tagged to it' }, status: :bad_request
+        else
+          @tag.destroy
+          render json: @tag, status: 204
```

Comment:
204 isn't appropriate here if we're returning the tag as JSON.  204 means "No Content"